clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -309,17 +300,18 @@ private static void writeInPlaceHeader(ByteBuffer buffer,
                                            long pid,
                                            short epoch,
                                            int sequence) {
-        buffer.putLong(OFFSET_OFFSET, offset);
-        buffer.putInt(SIZE_OFFSET, size - LOG_OVERHEAD);
-        buffer.put(MAGIC_OFFSET, magic);
-        buffer.put(ATTRIBUTES_OFFSET, attributes);
-        buffer.putLong(TIMESTAMP_OFFSET, timestamp);
-        buffer.putInt(OFFSET_DELTA_OFFSET, offsetDelta);
-        buffer.putLong(PID_OFFSET, pid);
-        buffer.putShort(EPOCH_OFFSET, epoch);
-        buffer.putInt(SEQUENCE_OFFSET, sequence);
-        long crc = Utils.computeChecksum(buffer, MAGIC_OFFSET, size - MAGIC_OFFSET);
-        buffer.putInt(CRC_OFFSET, (int) (crc & 0xffffffffL));
+        int position = buffer.position();
```

Comment:
It is a bug fix. Actually it was committed previously. I mistakenly pulled it here when I extracted the message format changes out of the PID mapping commit. Seemed like a pain to re-extract it from here, so I left it as it is. Should be fine due to the magic of squashing.